### PR TITLE
add event queue

### PR DESCRIPTION
**Pull request by balamurugana**
Signed-off-by: Bala.FA <bala@minio.io>

### CODE REVIEW

File: pkg/uevent/queue.go [hunk 1]
```diff
@@ -1,59 +1,129 @@
...
+	events       map[string]*deviceEvent
+	mutex        sync.Mutex
+	stopTimersCh chan struct{}
+	keyCh        chan string
+}
+
+func newEventQueue() *eventQueue {
+	return &eventQueue{
+		events: map[string]*deviceEvent{},
+		keyCh:  make(chan string, 16384),
```
Comment (Praveenrajmani) [inline pkg/uevent/queue.go:60]: do we need to set the buffer size to `16384`? can it be some lesser value perhaps?

File: pkg/uevent/queue.go [hunk 2]
```diff
@@ -1,59 +1,129 @@
...
-	
-	path := q.order[0]
-	q.order = q.order[1:]
-	if item, ok := q.items.LoadAndDelete(path); ok {
-		return item.(*deviceEvent)
+
+	return nil
+}
+
+func (e *eventQueue) pop() *deviceEvent {
```
Comment (Praveenrajmani) [inline pkg/uevent/queue.go:126]: can we add context handling in push and pop?

File: pkg/uevent/queue.go (hunk 1 above)
Comment (wlan0) [reply pkg/uevent/queue.go:60]: yeah, it can be smaller - just 64 or 128 should be enough

File: pkg/uevent/queue.go (hunk 2 above)
Comment (wlan0) [reply pkg/uevent/queue.go:126]: since there is a possibility of wait involved - context is needed here

**Pull request closed without merging**